include/exiv2/image_types.hpp

```diff
@@ -0,0 +1,56 @@
+// ***************************************************************** -*- C++ -*-
+/*
+ * Copyright (C) 2004-2019 Exiv2 authors
+ * This program is part of the Exiv2 distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, 5th Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef IMAGE_TYPES_H
+#define IMAGE_TYPES_H
+
+namespace Exiv2 {
+    /// Supported Image Formats
+    enum class ImageType{
```

Comment:
I know this is a big API change but as you already have mentioned the previous behaviour was buggy. 

The main problem I see is that it is not possible to keep the same numbers. Take a look here:
https://github.com/Exiv2/exiv2/pull/737/files#diff-5c74f56289aad5a5f43de672abf604a8L40

We had several types sharing the same value ... 🤦‍♂️ 

What I could do is to leave the `tiff` type with value 4, then continue with the other values that was not define in that file and put at the end of the Enum Type the types:

```
dng,
nef,
pef,
arw,
sr2,
srw,
```

What do you think?

Reply:
If we break the API, break it good so that people who have hardcoded the numbers will find out ASAP.

Reply:
After continuing working on this, I do not think users will suffer a lot about this change. In case they were using the previous `ImageType` namespace, they will need to change it for using the new scoped enum in the form of `ImageType::jpeg`. But that's all ... the **ImageType** type is not used in the API except for the **ImageFactory** class and not in the most common functions. 

I think we can keep that Enum class ordered alphabetically, and the numbers associated to each Image Type will not be very relevant.